### PR TITLE
Feature/g4 token proofs

### PR DESCRIPTION
**Pull request by zonotope**
This is a work in progress pull request to refator the gossip4 types into the messages repo so that we can use them to validate receive token transactions (related to https://github.com/quorumcontrol/messages/pull/34). 

I started down the path of wrapping the new messages type to cache their cbor representation in the app, but there is more left to be done to wrap the `RoundConfirmation` message type.

### CODE REVIEW

**Comment by zonotope** [comment]
Client tests are failing timing out, and I think the problem is in the round subscriber because the test nodes are all publishing their transactions. I suspect that the round subscriber is listening for the wrong pubsub messages, but I have not verified that.

After I fix the tests in the client module, I need to use [this function](https://github.com/quorumcontrol/tupelo/blob/feature/g4-token-proofs/gossip/proofvalidater.go#L40) To create a new validator generator that validate receive token transactions.

**Comment by tobowers** [comment]
i'm hoping that you pushed up all your commits, but the client tests are still relying on Proof so maybe not?

**Comment by zonotope** [comment]
:ping_pong:

File: proof/proof.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	isVerified, err := sigfuncs.Valid(ctx, sig, msg, nil)
+	if err != nil {
+		sp.SetTag("error", true)
+		return false, fmt.Errorf("error verifying round signature: %v", err)
+	}
+	sp.SetTag("verified", isVerified)
+	return isVerified, nil
+}
+
+func Verify(rootCtx context.Context, proof *gossip.Proof, quorumCount uint64, verKeys []*bls.VerKey) (bool, error) {
```
Comment (tobowers) [inline proof/proof.go]: no test here either? I guess if the client can/send/receive then that's enough testing here.

**Comment by tobowers** [comment]
conflict here as well.

File: proof/proof.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	if !validRoundConfirmation {
+		return false, nil
+	}
+
+	underlyingStore := nodestore.MustMemoryStore(ctx)
+	hamtStore := hamt.CborIpldStore{
+		Blocks: hamtwrapper.NewStore(underlyingStore),
+	}
+
+	roundCid, err := hamtStore.Put(ctx, proof.Round)
```
Comment (tobowers) [inline proof/proof.go]: noting that I think these CID calculations are incorrect vs what the client is doing.

Comment (zonotope) [reply proof/proof.go]: Well, that explains why the round and round confirmation cids aren't matching up in my tests!

Comment (tobowers) [reply proof/proof.go]: https://github.com/quorumcontrol/tupelo-go-sdk/pull/177/files#diff-67701a72352c0e31f086a1732e111565R15-R21

**Comment by zonotope** [comment]
@tobowers are you good with the tests now?

**Pull request merged**